File: Data Structures & Algorithms/meeting-schedule/submission-2.py
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
    def canAttendMeetings(self, intervals: List[Interval]) -> bool:
        intervals = sorted(intervals, key=lambda x: x.start)
        cur_end = 0

        for interval in intervals:
            start, end = interval.start, interval.end
            if start < cur_end:
                return False

            if cur_end < end:
                cur_end = end

        return True

File: Data Structures & Algorithms/non-overlapping-intervals/submission-2.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](auto& a, auto& b) {
            if (a[1] == b[1]) return a[0] < b[0];
            return a[1] < b[1];
        });

        int ret = 0;
        int end = INT_MIN;

        for (auto& interval : intervals) {
            int s = interval[0];
            int e = interval[1];

            if (s < end) {
                ret++;
            } else {
                end = e;
            }
        }
        return ret;
    }
};
